src/components/header/index.jsx

```diff
@@ -0,0 +1,162 @@
+import {
+  Box,
+  Stack,
+  Typography,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  ListItemIcon,
+  Divider,
+} from "@mui/material";
+import { FiSearch } from "react-icons/fi";
+import DefaultInput from "../inputs/default-input";
+import Logo from "../../assets/icons/logo.svg";
+import heartIcon from "../../assets/icons/heart-primary.svg";
+import userIcon from "../../assets/icons/user.svg";
+import bagIcon from "../../assets/icons/bag.svg";
+import NavLink from "../links/nav-link";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import ButtonWithIcon from "../buttons/button-with-Icon";
+import MenuIcon from "@mui/icons-material/Menu";
+import Drawer from "@mui/material/Drawer";
+import { useState } from "react";
+
+const Header = ({ Categories }) => {
+  const isLaptop = useMediaQuery((theme) => theme.breakpoints.down("lg"));
+  const isTablet = useMediaQuery((theme) => theme.breakpoints.down("md"));
+  const isMobile = useMediaQuery((theme) => theme.breakpoints.down("sm"));
+
+  const [mobileOpen, setMobileOpen] = useState(false);
+
+  const handleDrawerToggle = () => {
+    setMobileOpen((prevState) => !prevState);
+  };
+
+  const drawer = (
+    <List>
+      <ListItem>
+        <ListItemText primary={"Categories"} />
+      </ListItem>
+      <Divider />
+      {["Handbags", "Watches", "Skincare", "Jewellery", "Apperels"].map(
```

Comment:
It's better to store this array as a constant then use it here.

Reply:
Done